full_name.rb

```diff
@@ -0,0 +1,18 @@
+
+    full_name = []
+    print "What's your first name?"
+    first_name = gets.chomp.capitalize!
```

Comment:
```suggestion
    first_name = gets.chomp.capitalize
```

Ruby's string methods that end in `!` are typically for "in-place" replacements, and their behaviour is like this:

```ruby
foo = "bar"
foo.capitalize! # Notice how there is no assignment `foo =` on this line
puts foo # but it still prints "Bar" because the replacement is done "in-place"
```

In fact, the Ruby docs for [capitalize!](https://ruby-doc.org/core-3.0.3/String.html#method-i-capitalize-21) state:

> Returns `nil` if no changes are made.

So this is what I saw on the terminal when I ran your program when I entered "Daniel" for the first name, a blank middle name, and "S" for the last name:

```
What's your first name?Daniel
What's your middle name? (If you don't have a middle name, simply leave blank and hit enter.)
What's your last name?S
Hello!
```

Reply:
I can understand why you're auto-capitalizing the names we enter, but FYI in some cultures it's incorrect to capitalize every name; for example, Python creator Guido van Rossum, a Dutchman, has this to say on [his site](https://gvanrossum.github.io/):

> Dutch spelling rules dictate that when used in combination with my first name, "van" is not capitalized: "Guido van Rossum".

Reply:
very true on the name capitalization part. thanks for the feedback on the difference between .capitalize and capitalize! I could not figure out why I had to use unless nil instead of unless empty string but this feedback cleared up my issue!